recipe-client-addon/test/browser/browser_StudyStorage.js

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+Cu.import("resource://shield-recipe-client/lib/StudyStorage.jsm", this);
+
+// -------
+
+add_task(async function checkReset() {
+  ok(!StudyStorage.database, "StudyStorage starts without an IDB instance.");
+
+  await StudyStorage.getDatabase();
+  ok(StudyStorage.database, "StudyStorage creates an IDB instance.");
+
+  await StudyStorage.reset();
+  ok(!StudyStorage.database, "StudyStorage correctly removes its IDB instance.");
+});
+
+add_task(async function checkGetDatabase() {
+  await StudyStorage.reset(true);
+  ok(!StudyStorage.database, "StudyStorage starts without an IDB instance.");
+
+  const db1 = await StudyStorage.getDatabase();
+  ok(db1, "StudyStorage creates an IDB instance.");
+
+  const db2 = await StudyStorage.getDatabase();
+  is(db1, db2, "StudyStorage creates only one IDB instance at a time.");
+});
+
+
+add_task(async function checkSaveStudy() {
+  await StudyStorage.reset(true);
+  let success = true;
+
+  try {
+    await StudyStorage.saveStudyData({
+      name: "hi",
+      version: 2,
+      description: "fake",
+      id: 12345
+    });
+  } catch (err) {
+    success = false;
+    throw err;
```

Comment:
Since you're throwing here, the assertion below won't run anyway, so catching and saving the success status isn't necessary.